File: schemas/media-streaming/convex/saved_albums.ts
// saved_albums: albums saved to a user's personal library.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const saved_albums = defineTable({
  userId: v.id("users"),
  albumId: v.id("albums"),
})
  .index("by_user_id_album_id", ["userId", "albumId"])
  .index("by_user_id_creation_time", ["userId", "_creationTime"]);

File: schemas/media-streaming/spacetimedb/saved_albums.rs
// saved_albums: Albums saved to a user's personal library.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = saved_albums, public)]
pub struct SavedAlbum {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    pub album_id: String, // UUID — FK → albums.id (cascade delete)
    pub created_at: Timestamp,
}
// Composite unique: (user_id, album_id) — enforce in reducer logic
